### PR TITLE
[FVM] Benchstat CI improvements

### PR DESCRIPTION
**Pull request by janezpodhostnik**
1. Increased the number of bench runs to 10 (from 7) since other CI take longer so we have extra time to get some more accuracy on the benchmarks.
2. Disabled running regular tests during the benchmarks
3. There was a bug in one of the benchmarks causing it to error out
4. Added memstats to the benchmarking output

### CODE REVIEW

File: .github/workflows/bench.yml [hunk 1]
```diff
@@ -37,14 +37,14 @@ jobs:
 
       - name: Run benchmark on current branch
         run: |
-          (for i in {1..${{ steps.settings.outputs.benchmark_repetitions }}}; do go test ./fvm --bench . --tags relic -shuffle=on; done) | tee new.txt
+          (for i in {1..${{ steps.settings.outputs.benchmark_repetitions }}}; do go test ./fvm --bench . --tags relic -shuffle=on --benchmem --run ^$; done) | tee new.txt
```
Comment (janezpodhostnik) [inline .github/workflows/bench.yml:40]: Bench memory and don't run tests (just benchmarks).

**Review by turbolent — APPROVED** [review]
Nice! 

Could you please also port these improvements to the Cadence repo?

**Review by SaveTheRbtz — APPROVED** [review]

File: .github/workflows/bench.yml (hunk 1 above)
Comment (SaveTheRbtz) [reply .github/workflows/bench.yml:40]: why not just `-count ${{ steps.settings.outputs.benchmark_repetitions }}`

Comment (SaveTheRbtz) [reply .github/workflows/bench.yml:40]: (nit: you may also want to update the "The command `...` was used.")

Comment (janezpodhostnik) [reply .github/workflows/bench.yml:40]: > why not just -count ${{ steps.settings.outputs.benchmark_repetitions }}?

Last time I was checking this, the problem was that using `count + shuffle` didn't do what I wanted. 

count 3 + shuffle of tests {A, B, C} does:
- BBB AAA CCC

I wanted something more spread out, so this does:
- BCA CBA ACB

Which I think is statistically better if the runner is noisy.

**Comment by janezpodhostnik** [comment]
bors merge

**Pull request merged**